### PR TITLE
chirp: remove regex dependency

### PR DESCRIPTION
**Pull request by maisem**
Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

File: chirp/chirp.go [hunk 1]
```diff
@@ -89,7 +88,20 @@ func (b *BIRDClient) exec(cmd string, args ...interface{}) (string, error) {
 	return b.readResponse()
 }
 
-var respCodeRegex = regexp.MustCompile(`^\d{4}[ -]`)
+// hasResponseCode reports whether the provided byte slice is
+// prefixed with the BIRD response code.
```
Comment (bradfitz) [inline chirp/chirp.go:92]: "with a" ?

File: chirp/chirp.go [hunk 2]
```diff
@@ -89,7 +88,20 @@ func (b *BIRDClient) exec(cmd string, args ...interface{}) (string, error) {
...
+// Equivalent regex: `^\d{4}[ -]`.
+func hasResponseCode(s []byte) bool {
+	if len(s) < 5 {
+		return false
+	}
+	for i := 0; i < 4; i++ {
+		if s[i] < '0' || '9' < s[i] {
```
Comment (bradfitz) [inline chirp/chirp.go:99]: I think the normal Go style is the other way around, with the boundaries on the outside:

```go
   for _, b := range s[:4] {
         if '0' <= b && b <= '9' {
              continue
         }
         return false
    }
```

**Review by bradfitz — APPROVED** [review]

File: chirp/chirp.go (hunk 2 above)
Comment (maisem) [reply chirp/chirp.go]: Done

File: chirp/chirp.go (hunk 1 above)
Comment (maisem) [reply chirp/chirp.go]: Done

**Pull request merged**